### ISSUE TITLE
Nerfstudio Windows Installation Guide

### ISSUE THREAD

**Issue by itshanson** [nerfstudio-project/nerfstudio#3726]
### Steps to install Nerfstudio on Windows(Last updated 10 November 2025)

**Tools and Build Environment:**

- x64 Native Tools Command Prompt for VS 2022 LTSC 17.8
- GPU: RTX 4060 Laptop GPU
- Windows 11 Home
- Git

I have spent days on setting up my nerfstudio environment and this is what I learnt so far. I gained help from this page:

[https://github.com/nerfstudio-project/nerfstudio/issues/3600](url)

and Gemini 2.5 Pro. This a combined documentation of suggestions from both pages and my experiences. Please correct me if I am wrong by any means.

My nerfstudio environment has ddl conflicts and issues with colmap and ffmpeg packages. So I created a new environment for both documenting the installation processes and resolve the issues.

Prerequisites:

Install VS 2022 LTSC 17.8 here
https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-history
Make sure to have Desktop Development Tools with C++ installed.

You do not need to run the complex commands on nerfstudio official page, open the “x64 Native Tools Command Prompt  for VS 2022 LTSC 17.8”, do not run the commands on any other terminals like x65_x86 or x_86.

All of the following commands are ran in the "x64 Native Tools Command Prompt" terminal.

**1. Install Python 3.8**

`conda create --name nerfstudio --channel conda-forge python=3.8`

using forge to avoid ddl conflict(suggested by Gemini)

**2. Activate the environment**

conda activate nerfstudio

This step is necessary to avoid future ddl conflicts
```
conda config --env --add channels conda-forge
conda config --env --set channel_priority strict
```

**3. follow the steps on the official page to install torch**

`pip install torch==2.1.2+cu118 torchvision==0.16.2+cu118 --extra-index-url https://download.pytorch.org/whl/cu118
`
**4. Install Cuda11.8**
`conda install -c "nvidia/label/cuda-11.8.0" cuda-toolkit
`
**5. For future running nerfstudio**
```
pip install ninja
conda install colmap ffmpeg
```
(you do not need this for installing nerfstudio, but I ran into issues without these two packages when training)

**6. Make sure the conda version does not conflict with the system cuda.**

```
set TCNN_CUDA_ARCHITECTURES=89
set CUDA_HOME=%CONDA_PREFIX% 
```

**note: 89 is just for the RTX 40 series GPU, I am not sure about the other series so please look up or consult AI.**

Check your nvcc version is correct by running: `where.exe nvcc`
output:
```
C:\Users\xxxxx\miniconda3\envs\nerfstudio\bin\nvcc.exe 
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.9\bin\nvcc.exe
```
(if the miniconda version appears first then cuda11.8 is used in this virtual environment, tiny-cuda-nn installation requires specifically cuda11.8 in this case!!)

Check your build environment by running `where.exe cl`
If the nothing was found, then it means you are no running on the correct terminal

**7. Install tiny-cuda-nn**

`pip install git+https://github.com/NVlabs/tiny-cuda-nn/#subdirectory=bindings/torch
`

**8. If successful, run:**

`pip install nerfstudio
`
You might run into this:

```
Error: Collecting pywinpty>=2.0.1 (from jupyter-server<3,>=2.4.0->jupyterlab>=3.3.4->nerfstudio)

  Using cached pywinpty-2.0.14.tar.gz (27 kB)

  Installing build dependencies ... done

  Getting requirements to build wheel ... done

  Installing backend dependencies ... error

  error: subprocess-exited-with-error



  × pip subprocess to install backend dependencies did not run successfully.

  │ exit code: 1

  ╰─> [3 lines of output]

      ERROR: Ignored the following versions that require a different python version: 0.1.0 Requires-Python >=3.9; 0.1.1 Requires-Python >=3.9; 0.1.2 Requires-Python >=3.9; 0.1.3 Requires-Python >=3.9; 0.1.4 Requires-Python >=3.9; 0.1.5 Requires-Python >=3.9

      ERROR: Could not find a version that satisfies the requirement puccinialin (from versions: none)

      ERROR: No matching distribution found for puccinialin

      [end of output]



  note: This error originates from a subprocess, and is likely not a problem with pip.

error: subprocess-exited-with-error



× pip subprocess to install backend dependencies did not run successfully.

│ exit code: 1

╰─> See above for output.



note: This error originates from a subprocess, and is likely not a problem with pip. 
```

Error resolved after running: `pip install pywinpty==2.0.13`

**9. Go to “Train your first model”, and test if your installation is correct!**

Download some test data:
`ns-download-data nerfstudio --capture-name=poster
`
Train model
`ns-train nerfacto --data data/nerfstudio/poster
`

**Comment by jaxn-harvey** [comment]
This was helpful, thanks!

**Comment by vitordanielpereirasilva034-create** [comment]
Error: Collecting pywinpty>=2.0.1 (from jupyter-server<3,>=2.4.0->jupyterlab>=3.3.4->nerfstudio)

  Using cached pywinpty-2.0.14.tar.gz (27 kB)

  Installing build dependencies ... done

  Getting requirements to build wheel ... done

  Installing backend dependencies ... error

  error: subprocess-exited-with-error



  × pip subprocess to install backend dependencies did not run successfully.

  │ exit code: 1

  ╰─> [3 lines of output]

      ERROR: Ignored the following versions that require a different python version: 0.1.0 Requires-Python >=3.9; 0.1.1 Requires-Python >=3.9; 0.1.2 Requires-Python >=3.9; 0.1.3 Requires-Python >=3.9; 0.1.4 Requires-Python >=3.9; 0.1.5 Requires-Python >=3.9

      ERROR: Could not find a version that satisfies the requirement puccinialin (from versions: none)

      ERROR: No matching distribution found for puccinialin

      [end of output]



  note: This error originates from a subprocess, and is likely not a problem with pip.

error: subprocess-exited-with-error



× pip subprocess to install backend dependencies did not run successfully.

│ exit code: 1

╰─> See above for output.



note: This error originates from a subprocess, and is likely not a problem with pip.
